packages/components/src/form/FormInput.tsx

```diff
@@ -0,0 +1,184 @@
+
+import { appliedStyles, cns } from '@stcland/utils'
+
+import { FormInputStyles } from 'src/styles/componentTypes'
+import { nullFormContext } from './Form'
+import Label from './FormLabel'
+
+//*****************************************************************************
+// Interface
+//*****************************************************************************
+
+type TextAreaPropsShape = {
+  rows: number;
+};
+
+export type FormInputProps = {
+  name: string;
+  value?: any;
+  defaultValue?: any;
+  id?: string;
+  label?: string;
+  placeholder?: string;
+  valueAsNumber?: boolean;
+  required?: boolean;
+  disabled?: boolean;
+  hidden?: boolean;
+  touched?: boolean;
+  dirty?: boolean;
+  onFocus?: () => void;
+  textAreaProps?: TextAreaPropsShape;
+  customStyles?: Partial<FormInputStyles>;
+  onChange?: () => void;
+};
+
+const formInputProps = [
+  'type',
+  'name',
+  'value',
+  'label',
+  'placeholder',
+  'required',
+  'disabled',
+  'hidden',
+  'onFocus',
+  'customStyles',
+  'touched',
+  'dirty',
+] as const
+
+export function pickFormProps<T>(props: T): T {
+  return pick(formInputProps, props)
+}
+
+export enum FieldType {
+  TEXT = 'text',
+  TEXTAREA = 'textarea',
+  EMAIL = 'email',
+  CHECKBOX = 'checkbox',
+  'DATETIME-LOCAL' = 'datetime-local',
+  DATE = 'date',
+  TIME = 'time',
+  FILE = 'file',
+  NUMBER = 'number',
+  PASSWORD = 'password',
+  RADIO = 'radio',
+}
+
+type Props = FormInputProps & {
+  type: FieldType;
+  customStyles?: Partial<FormInputStyles>;
+};
+
+//*****************************************************************************
+// Components
+//*****************************************************************************
+
+const FormInput = ({
+  textAreaProps,
+  type,
+  name,
+  id,
+  value,
+  label = '',
+  defaultValue,
+  onFocus = () => {},
+  valueAsNumber = false,
+  disabled = false,
+  hidden = false,
+  placeholder = '',
+  required = false,
+  customStyles,
+}: Props) => {
+  const { register, setValue, clearErrors, formState, setFocus } =
+    useFormContext() || nullFormContext
+  const { errors } = formState || {}
+
+  useLayoutEffect(() => {
+    if (isNotNil(value)) {
+      setValue(name, value)
+    }
+  }, [name, value, setValue])
+
+  if (hidden) return null
+
+  const errorTextPath = [...name.split('.'), 'message']
+  const errorText: React.ReactNode = path(errorTextPath, errors)
+
+  const registerRequired = {
+    value: required,
+    message: 'is required',
+  }
+
+  const inputStyles: FormInputStyles = appliedStyles(
+    {
+      root: 'flex flex-col w-100 gap-2',
+      label: 'text-sm pt-2',
+      inputRoot: cns(
+        type !== FieldType.TEXTAREA && 'h-14',
+        'px-4 py-4 border-[#4B5563] border-solid rounded-lg bg-[#2A2C30] text-lg border text-white w-full dark:[color-scheme:dark]',
+        'disabled:text-zinc-400'
+      ),
+      error: 'text-red-300 italic text-sm',
+      info: '',
+    },
+    customStyles
+  )
```

Comment:
@ianmarquez couple of things here:

1. Can we use Tailwind variants instead of hex? `#4B5563 = gray-600` and for `#2A2C30` try using any similar from Tailwind color variants?

2. What would happen if `inputRoot` does not add a :disabled pseudo-class, adds a `replace` word at the beginning OR there are not default styles? The UI design would not be as expected... I think that we should have a disabled prop as part of the inputStyles, and add the pseudo class after the styles are merged:

`cn = {
  root: mergedStyles.root,
  label: mergedStyles.root,
  inputRoot: cns(mergedStyles.inputRoot, disabled:mergedStyles.disabled),
  ...rest
}`

Can you also add customStyles in the `stc.config.ts` of Storybook, so we can test all the `withStylesWrapper` variants with customizations?

Reply:
1st  and 3rd ask is done, however for the last one wouldn't that be the intended effect? the consumer can change everything that we expose, or are you seeing this as more of a guard rails thing wherein a consumer can only change very specific values.